// @andyet/simplewebrtc/components/PeerList.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const React = tslib_1.__importStar(require("react"));
const react_redux_1 = require("react-redux");
const Selectors_1 = require("../Selectors");
/**
 * @description
 *
 * @public
 *
 */
class PeerList extends React.Component {
    render() {
        const renderProps = {
            chatState: this.props.chatState || undefined,
            joinedCall: this.props.joinedCall || false,
            peers: this.props.peers || [],
            speaking: this.props.speaking || false
        };
        let render = this.props.render;
        if (!render && typeof this.props.children === 'function') {
            render = this.props.children;
        }
        return render ? render(renderProps) : React.createElement(React.Fragment, null, this.props.children);
    }
}
function mapStateToProps(state, props) {
    const filters = [
        'joinedCall',
        'speaking',
        'chatState',
        'requestingAttention'
    ];
    let peers = (0, Selectors_1.getPeersForRoom)(state, props.room);
    if (props.includeSelf) {
        const user = (0, Selectors_1.getUserDataForRoom)(state, props.room);
        console.log('User', user);
        if (user) {
            peers.unshift(user);
        }
    }
    peers = peers.filter((peer) => {
        for (const filter of filters) {
            if (props[filter] !== undefined && peer[filter] !== props[filter]) {
                return false;
            }
        }
        return true;
    });
    if (props.filter) {
        peers = peers.filter(props.filter);
    }
    if (props.sort) {
        peers = peers.sort(props.sort);
    }
    return {
        ...props,
        peers
    };
}
exports.default = (0, react_redux_1.connect)(mapStateToProps)(PeerList);
