// @andyet/simplewebrtc/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createStore = exports.screensharing = exports.isSupportedBrowser = exports.reducer = exports.VolumeMeter = exports.Video = exports.UserControls = exports.StayDownContainer = exports.Selectors = exports.Room = exports.RequestUserMedia = exports.RequestDisplayMedia = exports.RemoteMediaList = exports.RemoteAudioPlayer = exports.Provider = exports.PeerList = exports.PeerControls = exports.NotSupported = exports.Notifications = exports.NotConnected = exports.MediaControls = exports.LocalMediaList = exports.GridLayout = exports.Failed = exports.Disconnected = exports.DeviceList = exports.Connecting = exports.Connected = exports.Commands = exports.ChatList = exports.ChatInputTextArea = exports.ChatInput = exports.ChatComposers = exports.Audio = exports.Actions = void 0;
const tslib_1 = require("tslib");
const redux_1 = require("redux");
const redux_thunk_1 = tslib_1.__importDefault(require("redux-thunk"));
const Actions = tslib_1.__importStar(require("./actions"));
exports.Actions = Actions;
const ScreenSharingHelpers = tslib_1.__importStar(require("./lib/Screensharing"));
const reducers_1 = tslib_1.__importDefault(require("./reducers"));
Object.defineProperty(exports, "reducer", { enumerable: true, get: function () { return reducers_1.default; } });
const Selectors = tslib_1.__importStar(require("./Selectors"));
exports.Selectors = Selectors;
const Audio_1 = tslib_1.__importDefault(require("./components/Audio"));
exports.Audio = Audio_1.default;
const ChatComposers_1 = tslib_1.__importDefault(require("./components/ChatComposers"));
exports.ChatComposers = ChatComposers_1.default;
const ChatInput_1 = tslib_1.__importStar(require("./components/ChatInput"));
exports.ChatInput = ChatInput_1.default;
Object.defineProperty(exports, "ChatInputTextArea", { enumerable: true, get: function () { return ChatInput_1.ChatInputTextArea; } });
const ChatList_1 = tslib_1.__importDefault(require("./components/ChatList"));
exports.ChatList = ChatList_1.default;
const Commands_1 = tslib_1.__importDefault(require("./components/Commands"));
exports.Commands = Commands_1.default;
const DeviceList_1 = tslib_1.__importDefault(require("./components/DeviceList"));
exports.DeviceList = DeviceList_1.default;
const GridLayout_1 = tslib_1.__importDefault(require("./components/GridLayout"));
exports.GridLayout = GridLayout_1.default;
const LocalMediaList_1 = tslib_1.__importDefault(require("./components/LocalMediaList"));
exports.LocalMediaList = LocalMediaList_1.default;
const MediaControls_1 = tslib_1.__importDefault(require("./components/MediaControls"));
exports.MediaControls = MediaControls_1.default;
const Notifications_1 = tslib_1.__importDefault(require("./components/Notifications"));
exports.Notifications = Notifications_1.default;
const PeerControls_1 = tslib_1.__importDefault(require("./components/PeerControls"));
exports.PeerControls = PeerControls_1.default;
const PeerList_1 = tslib_1.__importDefault(require("./components/PeerList"));
exports.PeerList = PeerList_1.default;
const Provider_1 = tslib_1.__importStar(require("./components/Provider"));
exports.Provider = Provider_1.default;
Object.defineProperty(exports, "Connected", { enumerable: true, get: function () { return Provider_1.Connected; } });
Object.defineProperty(exports, "Connecting", { enumerable: true, get: function () { return Provider_1.Connecting; } });
Object.defineProperty(exports, "Disconnected", { enumerable: true, get: function () { return Provider_1.Disconnected; } });
Object.defineProperty(exports, "Failed", { enumerable: true, get: function () { return Provider_1.Failed; } });
Object.defineProperty(exports, "NotConnected", { enumerable: true, get: function () { return Provider_1.NotConnected; } });
Object.defineProperty(exports, "NotSupported", { enumerable: true, get: function () { return Provider_1.NotSupported; } });
const RemoteAudioPlayer_1 = tslib_1.__importDefault(require("./components/RemoteAudioPlayer"));
exports.RemoteAudioPlayer = RemoteAudioPlayer_1.default;
const RemoteMediaList_1 = tslib_1.__importDefault(require("./components/RemoteMediaList"));
exports.RemoteMediaList = RemoteMediaList_1.default;
const RequestDisplayMedia_1 = tslib_1.__importDefault(require("./components/RequestDisplayMedia"));
exports.RequestDisplayMedia = RequestDisplayMedia_1.default;
const RequestUserMedia_1 = tslib_1.__importDefault(require("./components/RequestUserMedia"));
exports.RequestUserMedia = RequestUserMedia_1.default;
const Room_1 = tslib_1.__importDefault(require("./components/Room"));
exports.Room = Room_1.default;
const StayDownContainer_1 = tslib_1.__importDefault(require("./components/StayDownContainer"));
exports.StayDownContainer = StayDownContainer_1.default;
const UserControls_1 = tslib_1.__importDefault(require("./components/UserControls"));
exports.UserControls = UserControls_1.default;
const Video_1 = tslib_1.__importDefault(require("./components/Video"));
exports.Video = Video_1.default;
const VolumeMeter_1 = tslib_1.__importDefault(require("./components/VolumeMeter"));
exports.VolumeMeter = VolumeMeter_1.default;
exports.isSupportedBrowser = Selectors.isSupportedBrowser();
exports.screensharing = ScreenSharingHelpers;
const initial = {
    simplewebrtc: {}
};
function createStore() {
    return (0, redux_1.createStore)((0, redux_1.combineReducers)({
        simplewebrtc: reducers_1.default
    }), initial, (0, redux_1.compose)((0, redux_1.applyMiddleware)(redux_thunk_1.default)));
}
exports.createStore = createStore;
